kogito-codegen/src/main/java/org/kie/kogito/codegen/process/ProcessGenerator.java

```diff
@@ -382,76 +343,113 @@ public ClassOrInterfaceDeclaration classDeclaration() {
+                constructorBody
+                    .addStatement(
+                                  new MethodCallExpr(new NameExpr(WORK_ITEM_MANAGER), "registerWorkItemHandler")
+                                      .addArgument(new StringLiteralExpr(handler.getKey()))
+                                      .addArgument(new NameExpr(varName)));
+                handler.getValue().findAll(FieldDeclaration.class).forEach(fd -> forEachField(clazz, fd));
+                additionalClasses.add(handler.getValue());
+            }
+        }
         
-        ProcessMetaData processMetaData = processGenerator.generate();
+        constructorBody.addStatement(new MethodCallExpr(new ThisExpr(), "activate"));
 
         cls.addExtendedType(abstractProcessType(modelTypeName))
-                .addMember(fieldDeclaration)
-                .addMember(emptyConstructorDeclaration)
+                .addMember(new FieldDeclaration()
+                           .addVariable(new VariableDeclarator(new ClassOrInterfaceType(null, appCanonicalName), APPLICATION)))
                 .addMember(constructorDeclaration)
+                .addMember(getConstructorDeclaration())
                 .addMember(createInstanceMethod(processInstanceFQCN))
                 .addMember(createInstanceWithBusinessKeyMethod(processInstanceFQCN))
-                .addMember(createModelMethod)
+                .addMember(new MethodDeclaration()
+                           .addModifier(Keyword.PUBLIC)
+                           .setName(CREATE_MODEL)
+                           .setType(modelTypeName)
+                           .addAnnotation(Override.class)
+                           .setBody(new BlockStmt()
+                                    .addStatement(new ReturnStmt(new ObjectCreationExpr(null, 
+                                                                                        new ClassOrInterfaceType(null, modelTypeName), 
+                                                                                        NodeList.nodeList())))))
                 .addMember(createInstanceGenericMethod(processInstanceFQCN))
                 .addMember(createInstanceGenericWithBusinessKeyMethod(processInstanceFQCN))
                 .addMember(createInstanceGenericWithWorkflowInstanceMethod(processInstanceFQCN))
                 .addMember(createReadOnlyInstanceGenericWithWorkflowInstanceMethod(processInstanceFQCN))
-                .addMember(internalConfigure(processMetaData))
-                .addMember(internalRegisterListeners(processMetaData))
                 .addMember(process(processMetaData));
+
         
+        MethodDeclaration configure = internalConfigure(processMetaData);
+        if (configure != null) {
+            cls.addMember(configure);
+        }
+        MethodDeclaration registerListener = internalRegisterListeners(processMetaData);
+        if (registerListener != null) {
+            cls.addMember(registerListener);
+        }
```

Comment:
If you make them return `Optional<MethodDeclaration>` you can replace with something like 
```suggestion
        internalConfigure(processMetaData).ifPresent(cls::addMember);
        internalRegisterListeners(processMetaData)::ifPresent(cls::addMember);
```